runtime/stackmap/stackmap.c

```diff
@@ -436,7 +436,7 @@ mapStack(UDATA *scratch, UDATA totalStack, U_8 * map, J9ROMClass * romClass, J9R
 					J9ROMNAMEANDSIGNATURE_SIGNATURE(J9ROMFIELDREF_NAMEANDSIGNATURE
 													((J9ROMFieldRef *) (&(pool[index]))));
 				signature = (U_8) J9UTF8_DATA(utf8Signature)[0];
-				if ((signature == 'L') || (signature == '[')) {
+				if (J9_IS_OBJECT_OR_VALUETYPE(signature) || ('[' == signature)) {
```

Comment:
The `PUSH(OBJ)` here seems suspect for a Q type

Reply:
This seems like a case where the present (incomplete) implementation of Q types as nearly regular objects works, but the future flattenable implementation will not. I have added a comment pointing to #4083, and a note there.